core/bubbles/bubble.ts

```diff
@@ -0,0 +1,558 @@
+    this.tail = dom.createSvgElement(Svg.PATH, {}, embossGroup);
+    this.background = dom.createSvgElement(
+        Svg.RECT, {
+          'class': 'blocklyDraggable',
+          'x': 0,
+          'y': 0,
+          'rx': Bubble.BORDER_WIDTH,
+          'ry': Bubble.BORDER_WIDTH,
+        },
+        embossGroup);
+    this.contentContainer = dom.createSvgElement(Svg.G, {}, this.svgRoot);
+
+    browserEvents.conditionalBind(
+        this.background, 'pointerDown', this, this.onMouseDown);
+  }
+
+  /** Dispose of this bubble. */
+  dispose() {
+    dom.removeNode(this.svgRoot);
+    this.disposed = true;
+  }
+
+  /**
+   * Set the location the tail of this bubble points to.
+   *
+   * @param anchor The location the tail of this bubble points to.
+   * @param relayout If true, reposition the bubble from scratch so that it is
+   *     optimally visible. If false, reposition it so it maintains the same
+   *     position relative to the anchor.
+   */
+  setAnchorLocation(anchor: Coordinate, relayout = false) {
+    this.anchor = anchor;
+    if (relayout) {
+      this.positionByRect(this.ownerRect);
+    } else {
+      this.positionRelativeToAnchor();
+    }
+    this.renderTail();
+  }
+
+  /** Sets the position of this bubble relative to its anchor. */
+  setPositionRelativeToAnchor(left: number, top: number) {
+    this.relativeLeft = left;
+    this.relativeTop = top;
+    this.positionRelativeToAnchor();
+    this.renderTail();
+  }
+
+  /** @return the size of this bubble. */
+  protected getSize() {
+    return this.size;
+  }
+
+  /**
+   * Sets the size of this bubble, including the border.
+   *
+   * @param size Sets the size of this bubble, including the border.
+   * @param relayout If true, reposition the bubble from scratch so that it is
+   *     optimally visible. If false, reposition it so it maintains the same
+   *     position relative to the anchor.
+   */
+  protected setSize(size: Size, relayout = false) {
+    // TODO: set size.
+    size.width = Math.max(size.width, Bubble.MIN_SIZE);
+    size.height = Math.max(size.height, Bubble.MIN_SIZE);
+    this.size = size;
+
+    this.background.setAttribute('width', `${size.width}`);
+    this.background.setAttribute('height', `${size.height}`);
+
+    if (relayout) {
+      this.positionByRect(this.ownerRect);
+    } else {
+      this.positionRelativeToAnchor();
+    }
+    this.renderTail();
+  }
+
+  /** Returns the colour of the background and tail of this bubble. */
+  protected getColour(): string {
+    return this.colour;
+  }
+
+  /** Sets the colour of the background and tail of this bubble. */
+  protected setColour(colour: string) {
+    this.colour = colour;
+    this.tail.setAttribute('fill', colour);
+    this.background.setAttribute('fill', colour);
+  }
+
+  /** Gets the emboss filter for this bubble. */
+  private getFilter(): {filter?: string} {
+    // TODO: Do we think this is actually still a problem??
```

Comment:
this was fun to look into.... i could only find a reference to this being a problem in java 8 in 2014. i'd say remove it